src/cargo/core/resolver/mod.rs

```diff
@@ -148,8 +150,19 @@ pub fn resolve(
         Some(config) => config.cli_unstable().direct_minimal_versions,
         None => false,
     };
-    let mut registry =
-        RegistryQueryer::new(registry, replacements, version_prefs, minimal_versions);
+    if !config
+        .map(|c| c.cli_unstable().msrv_policy)
+        .unwrap_or(false)
+    {
+        max_rust_version = None;
+    }
```

Comment:
I centralized the feature check to this one spot

I am somewhat tempted to do something similar for `honor_rust_version` and have the config initialized with it and just check the config here.